compiler/x/codegen/X86BinaryEncoding.cpp

```diff
@@ -176,21 +176,36 @@ uint8_t getMemoryBarrierBinaryLengthLowerBound(int32_t barrier, TR::CodeGenerato
 
 // -----------------------------------------------------------------------------
 // OMR::X86::Instruction:: member functions
+bool OMR::X86::Instruction::needsLockPrefix()
+   {
+   return self()->getOpCode().needsLockPrefix();
+   }
 
-uint8_t *OMR::X86::Instruction::generateBinaryEncoding()
+uint8_t* OMR::X86::Instruction::generateBinaryEncoding()
    {
-   // *this    swipeable for debugging purposes
-   uint8_t *instructionStart = self()->cg()->getBinaryBufferCursor();
-   uint8_t *cursor           = instructionStart;
-   if (self()->getOpCode().needsRepPrefix())
+   for(;;)
       {
-      *cursor++ = 0xf3;
+      uint8_t *instructionStart = self()->cg()->getBinaryBufferCursor();
+      uint8_t *cursor           = instructionStart;
+      self()->setBinaryEncoding(instructionStart);
+      if (self()->getOpCode().needsRepPrefix())
+         {
+         *cursor++ = 0xf3;
+         }
+      if (self()->needsLockPrefix())
+         {
+         *cursor++ = 0xf0;
+         }
+      cursor = self()->generateRepeatedRexPrefix(cursor);
+      cursor = self()->getOpCode().binary(cursor, self()->rexBits());
+      cursor = self()->generateOperand(cursor);
+      if (cursor)
```

Comment:
When can cursor be false - it would seem like you would have generated instructions into the stream but we would fail to set th einstruction length? What do we return if the cursor does become NULL?